common/rpc/interceptor/retry.go

```diff
@@ -0,0 +1,68 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package interceptor
+
+import (
+	"context"
+
+	"go.temporal.io/server/common/backoff"
+	"google.golang.org/grpc"
+)
+
+type (
+	RetryableInterceptor struct {
+		policy      backoff.RetryPolicy
+		isRetryable backoff.IsRetryable
+	}
+)
+
+var _ grpc.UnaryServerInterceptor = (*RetryableInterceptor)(nil).Intercept
```

Comment:
This is great.. are you planning to consolidate the "retryable rpc clients" (`client/*/retryable_client*.go`) with an client interceptor also?

Reply:
Oh that's a good idea. I wasn't planning for it, but yeah I can replace them with a client interceptor.